src/components/anchor-navigation/anchor-navigation.component.js

```diff
@@ -0,0 +1,160 @@
+import React, {
+  useState, useRef, useEffect, useCallback
+} from 'react';
+import PropTypes from 'prop-types';
+import throttle from 'lodash/throttle';
+
+import Event from '../../utils/helpers/events';
+import { StyledAnchorNavigation, StyledNavigation, StyledContent } from './anchor-navigation.style';
+import AnchorNavigationItem from './anchor-navigation-item.component';
+
+const SECTION_VISIBILITY_OFFSET = 200;
+const SCROLL_THROTTLE = 100;
+
+const AnchorNavigation = ({ children, stickyNavigation, styleOverride }) => {
+  const [selectedIndex, setSelectedIndex] = useState(0);
+
+  const sectionRefs = useRef(React.Children.map(stickyNavigation.props.children, child => child.props.target));
+
+  const anchorRefs = useRef(Array.from({
+    length: React.Children.count(stickyNavigation.props.children)
+  }, () => React.createRef()));
+
+  const contentRef = useRef();
+
+  const navigationRef = useRef();
+
+  const isUserScroll = useRef(true);
+
+  const isUserScrollTimer = useRef();
+
+  const setSelectedAnchorBasedOnScroll = useCallback(() => {
+    const sectionsTopOffsets = sectionRefs.current.map(({ current }) => current.getBoundingClientRect().top);
+    const { top: navTopOffset } = navigationRef.current.getBoundingClientRect();
+
+    const indexOfSmallestNegativeTopOffset = sectionsTopOffsets.reduce((currentTopIndex, sectionTopOffset, index) => {
+      if (sectionTopOffset - SECTION_VISIBILITY_OFFSET > navTopOffset) return currentTopIndex;
+      return sectionTopOffset > sectionsTopOffsets[currentTopIndex] ? index : currentTopIndex;
+    }, 0);
+
+    setSelectedIndex(indexOfSmallestNegativeTopOffset);
+  }, []);
```

Comment:
you should decide to either drop useCallback or specify the dependencies: https://reactjs.org/docs/hooks-reference.html#usecallback

Reply:
It is not necessary to specify dependencies here because `indexOfSmallestNegativeTopOffset` only takes `sectionRefs` and `navigationRef` from the outer scope and both of these are created by useRef meaning that they keep their references through whole lifecycle of the component.

Adding `[sectionRefs, navigationRef]` wont change anything in this case

Reply:
in fact I was wrong, due to how ref callbacks work we can have empty array of dependencies in this case: https://reactjs.org/docs/hooks-faq.html